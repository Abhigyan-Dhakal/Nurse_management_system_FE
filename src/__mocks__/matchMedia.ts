export default Object.defineProperty(window, "matchMedia", {
  value: () => {
    return {
      matches: false,
      addListener: () => {},
      removeListener: () => {},
    };
  },
});
