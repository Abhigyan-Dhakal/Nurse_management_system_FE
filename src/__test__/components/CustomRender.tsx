import type { PreloadedState } from "@reduxjs/toolkit";
import { configureStore } from "@reduxjs/toolkit";
import type { RenderOptions } from "@testing-library/react";
import { render } from "@testing-library/react";
import React, { PropsWithChildren } from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { allReducers, RootState, store } from "../../redux/store";
export const setupStore = (preloadedState?: PreloadedState<RootState>) => {
  return configureStore({
    reducer: allReducers,
    preloadedState,
  });
};
interface ExtendedRenderOptions extends Omit<RenderOptions, "queries"> {
  preloadedState?: PreloadedState<RootState>;
  store?: typeof store;
}
export function renderWithProvidersAndNavigation(
  ui: React.ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return (
      <Provider store={store}>
        <BrowserRouter>{children}</BrowserRouter>
      </Provider>
    );
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
