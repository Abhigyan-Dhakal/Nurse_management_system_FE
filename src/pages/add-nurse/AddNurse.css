.add-nurse-container {
  max-width: 760px;
  margin: 0px auto;
  padding: 40px 80px;
}

.add-nurse-heading {
  font-size: 42px;
  color: rgb(29, 84, 156);
}
