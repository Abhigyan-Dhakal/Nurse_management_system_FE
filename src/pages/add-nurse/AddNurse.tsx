import React from "react";
import { AddNurseForm } from "../../components";
import "./AddNurse.css";

type Props = {};

export const AddNurse = (props: Props) => {
  return (
    <div className="add-nurse-container">
      <div>
        <h1 className="add-nurse-heading">Add new Nurse details</h1>
      </div>
      <div>
        <AddNurseForm />
      </div>
    </div>
  );
};
