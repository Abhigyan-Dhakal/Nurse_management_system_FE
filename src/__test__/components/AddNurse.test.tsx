import "../../__mocks__/matchMedia";
import { AddNurse } from "../../pages/add-nurse/AddNurse";
import { renderWithProvidersAndNavigation } from "./CustomRender";

it("should render adding nurse details screen properly", () => {
  const page = renderWithProvidersAndNavigation(<AddNurse />);
  expect(page.asFragment()).toMatchSnapshot();
});
