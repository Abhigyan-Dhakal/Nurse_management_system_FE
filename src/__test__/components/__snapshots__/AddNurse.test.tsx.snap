// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render adding nurse details screen properly 1`] = `
<DocumentFragment>
  <div
    class="add-nurse-container"
  >
    <div>
      <h1
        class="add-nurse-heading"
      >
        Add new Nurse details
      </h1>
    </div>
    <div>
      <form
        autocomplete="off"
        class="ant-form ant-form-horizontal"
        id="basic"
      >
        <div>
          <p
            class="form-label"
          >
            Name 
            <span
              class="compulsory"
            >
              *
            </span>
          </p>
        </div>
        <div
          class="ant-form-item"
        >
          <div
            class="ant-row ant-form-item-row"
          >
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <input
                    aria-required="true"
                    class="ant-input"
                    id="basic_name"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <p
            class="form-label"
          >
            Email
          </p>
        </div>
        <div
          class="ant-form-item"
        >
          <div
            class="ant-row ant-form-item-row"
          >
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <input
                    class="ant-input"
                    id="basic_email"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <p
            class="form-label"
          >
            Working Days 
            <span
              class="compulsory"
            >
              *
            </span>
          </p>
        </div>
        <div
          class="ant-form-item"
        >
          <div
            class="ant-row ant-form-item-row"
          >
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <div
                    class="ant-input-number ant-input-number-in-form-item"
                    style="width: 100%;"
                  >
                    <div
                      class="ant-input-number-handler-wrap"
                    >
                      <span
                        aria-disabled="false"
                        aria-label="Increase Value"
                        class="ant-input-number-handler ant-input-number-handler-up"
                        role="button"
                        unselectable="on"
                      >
                        <span
                          aria-label="up"
                          class="anticon anticon-up ant-input-number-handler-up-inner"
                          role="img"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="up"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 00140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
                            />
                          </svg>
                        </span>
                      </span>
                      <span
                        aria-disabled="false"
                        aria-label="Decrease Value"
                        class="ant-input-number-handler ant-input-number-handler-down"
                        role="button"
                        unselectable="on"
                      >
                        <span
                          aria-label="down"
                          class="anticon anticon-down ant-input-number-handler-down-inner"
                          role="img"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="down"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                            />
                          </svg>
                        </span>
                      </span>
                    </div>
                    <div
                      class="ant-input-number-input-wrap"
                    >
                      <input
                        aria-required="true"
                        aria-valuemax="30"
                        aria-valuemin="1"
                        autocomplete="off"
                        class="ant-input-number-input"
                        id="basic_workingDays"
                        role="spinbutton"
                        step="1"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <p
            class="form-label"
          >
            Duty Start Time 
            <span
              class="compulsory"
            >
              *
            </span>
          </p>
        </div>
        <div
          class="ant-form-item"
        >
          <div
            class="ant-row ant-form-item-row"
          >
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <div
                    class="ant-picker"
                    style="width: 100%;"
                  >
                    <div
                      class="ant-picker-input"
                    >
                      <input
                        aria-required="true"
                        autocomplete="off"
                        id="basic_dutyStartTime"
                        placeholder="Select time"
                        readonly=""
                        size="10"
                        title=""
                        value=""
                      />
                      <span
                        class="ant-picker-suffix"
                      >
                        <span
                          aria-label="clock-circle"
                          class="anticon anticon-clock-circle"
                          role="img"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="clock-circle"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                            />
                            <path
                              d="M686.7 638.6L544.1 535.5V288c0-4.4-3.6-8-8-8H488c-4.4 0-8 3.6-8 8v275.4c0 2.6 1.2 5 3.3 6.5l165.4 120.6c3.6 2.6 8.6 1.8 11.2-1.7l28.6-39c2.6-3.7 1.8-8.7-1.8-11.2z"
                            />
                          </svg>
                        </span>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <p
            class="form-label"
          >
            Duty End Time 
            <span
              class="compulsory"
            >
              *
            </span>
          </p>
        </div>
        <div
          class="ant-form-item"
        >
          <div
            class="ant-row ant-form-item-row"
          >
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <div
                    class="ant-picker"
                    style="width: 100%;"
                  >
                    <div
                      class="ant-picker-input"
                    >
                      <input
                        aria-required="true"
                        autocomplete="off"
                        id="basic_dutyEndTime"
                        placeholder="Select time"
                        readonly=""
                        size="10"
                        title=""
                        value=""
                      />
                      <span
                        class="ant-picker-suffix"
                      >
                        <span
                          aria-label="clock-circle"
                          class="anticon anticon-clock-circle"
                          role="img"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="clock-circle"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                            />
                            <path
                              d="M686.7 638.6L544.1 535.5V288c0-4.4-3.6-8-8-8H488c-4.4 0-8 3.6-8 8v275.4c0 2.6 1.2 5 3.3 6.5l165.4 120.6c3.6 2.6 8.6 1.8 11.2-1.7l28.6-39c2.6-3.7 1.8-8.7-1.8-11.2z"
                            />
                          </svg>
                        </span>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <p
            class="form-label"
          >
            Address 
            <span
              class="compulsory"
            >
              *
            </span>
          </p>
        </div>
        <div
          class="ant-form-item"
        >
          <div
            class="ant-row ant-form-item-row"
          >
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <input
                    aria-required="true"
                    class="ant-input"
                    id="basic_address"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <p
            class="form-label"
          >
            Contact 
            <span
              class="compulsory"
            >
              *
            </span>
          </p>
        </div>
        <div
          class="ant-form-item"
        >
          <div
            class="ant-row ant-form-item-row"
          >
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <input
                    aria-required="true"
                    class="ant-input"
                    id="basic_contact"
                    type="number"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <p
            class="form-label"
          >
            Rounding Manager
          </p>
        </div>
        <div
          class="ant-form-item"
        >
          <div
            class="ant-row ant-form-item-row"
          >
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <button
                    aria-checked="false"
                    class="ant-switch"
                    id="basic_isRoundingManager"
                    role="switch"
                    type="button"
                  >
                    <div
                      class="ant-switch-handle"
                    />
                    <span
                      class="ant-switch-inner"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <p
            class="form-label"
          >
            Picture 
          </p>
        </div>
        <div
          class="ant-form-item"
        >
          <div
            class="ant-row ant-form-item-row"
          >
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <span>
                    <div
                      class="ant-upload ant-upload-select ant-upload-select-text"
                    >
                      <span
                        class="ant-upload"
                        role="button"
                        tabindex="0"
                      >
                        <input
                          accept=""
                          id="basic_picture"
                          style="display: none;"
                          type="file"
                        />
                        <button
                          class="ant-btn ant-btn-default"
                          type="button"
                        >
                          <span
                            aria-label="upload"
                            class="anticon anticon-upload"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="upload"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M400 317.7h73.9V656c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V317.7H624c6.7 0 10.4-7.7 6.3-12.9L518.3 163a8 8 0 00-12.6 0l-112 141.7c-4.1 5.3-.4 13 6.3 13zM878 626h-60c-4.4 0-8 3.6-8 8v154H214V634c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v198c0 17.7 14.3 32 32 32h684c17.7 0 32-14.3 32-32V634c0-4.4-3.6-8-8-8z"
                              />
                            </svg>
                          </span>
                          <span>
                            Click to Upload
                          </span>
                        </button>
                      </span>
                    </div>
                    <div
                      class="ant-upload-list ant-upload-list-text"
                    />
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-form-item button--form-item"
        >
          <div
            class="ant-row ant-form-item-row"
          >
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <button
                    class="ant-btn ant-btn-primary ant-btn-lg ant-btn-block"
                    type="submit"
                  >
                    <span>
                      Edit Nurse
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
